### PR TITLE
added __repr__ to NDUncertainty.

### PR DESCRIPTION
**Pull request by MSeifert04**
I think ``NDUncertainty`` (and subclasses) should have a basic `__repr__` method instead of displaying the Memory-Adress:
```
>>> ndd = NDData(np.array([[1,2,3],[2,3,4]])) # no specified unit
>>> ndd.uncertainty = np.ones((10,10))
INFO: Uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
>>> ndd.uncertainty
UnknownUncertainty([[ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                    [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                    [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                    [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                    [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                    [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                    [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                    [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                    [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                    [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.]])
```
----

I'm not sure if the unit should be displayed and I removed it again from the PR since `NDData` doesn't display it's unit. Also I've noticed some problems with the unit because it grabs the unit from the parent if no unit itself is set. This made it impossible viewing an uncertainty without unit and without parent.

Example with unit:
```
>>> ndd = NDData(np.array([[1,2,3],[2,3,4]]), unit='m')
>>> ndd.uncertainty = StdDevUncertainty(np.ones((10,10)))
>>> ndd.uncertainty
StdDevUncertainty([[ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                   [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                   [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                   [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                   [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                   [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                   [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                   [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                   [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.],
                   [ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.]], unit: m)

### CODE REVIEW

**Comment by MSeifert04** [comment]
I think the test failure is unrelated to this PR.

**Comment by bsipocz** [comment]
I've restarted the appveyor build.

**Comment by pllim** [comment]
Probably needs a change log. I'd say for "API change" in 1.2?

**Comment by MSeifert04** [comment]
Included and also small doctests for the representation. I think there is no need for explicit unit-tests.

**Comment by pllim** [comment]
:+1: Looks good to me!

**Comment by MSeifert04** [comment]
I've removed the unit from the `__repr__`. It did raise an Error when the viewing an uncertainty without unit and without parent.

File: CHANGES.rst [hunk 1]
```diff
@@ -171,6 +171,8 @@ API changes
   - ``NDUncertainty``: Public methods ``propagate_add``, etc. are replaced by
     a general ``propagate`` method. [#4272]
 
+  - ``NDUncertainty``: Does implements a meaningful ``repr``. [#4787]
```
Comment (mwcraig) [inline CHANGES.rst]: Minor change: Remove `Does` or change to `Does implement`.

**Comment by mwcraig** [comment]
Except for a really minor wording change in the changelog this is good.

**Comment by MSeifert04** [comment]
@mwcraig corrected. Thanks for the review. :+1:

**Comment by MSeifert04** [comment]
Don't merge this just yet. I've encountered a problem here if the ``array`` is not a numpy-array ... I'll submit a fix later.

**Comment by MSeifert04** [comment]
The AppVeyor fail is unrelated to the changes.

**Comment by mwcraig** [comment]
Merging...

**Pull request merged**